src/AgGrid/ValueFormat.elm

```diff
@@ -0,0 +1,205 @@
+
+
+# filterValueGetter
+
+@docs currencyFilterValueGetter, decimalFilterValueGetter, percentFilterValueGetter
+
+-}
+
+import String.Interpolate
+
+
+
+-- VALUE FORMATTER
+
+
+{-| Format a cell value as localized CURRENCY value utilizing the Javascript `Intl.NumberFormat` class.
+
+    cellValue = "1200.35"
+
+    currencyValueFormatter { currency = "EUR", countryCode = "de-DE" }
+    > "1.200,35 €"
+
+    currencyValueFormatter { currency = "EUR", countryCode = "en-US" }
+    > "€1,200.35"
+
+-}
+currencyValueFormatter : { currency : String, countryCode : String } -> String
+currencyValueFormatter { currency, countryCode } =
+    String.Interpolate.interpolate """
+        let input;
+
+        if (!value) { return null; }
+
+        if (typeof value === 'object' && value.hasOwnProperty('value')) {
+            // Group value
+            input = value.value
+        } else {
+            // Cell value
+            input = value
+        }
+
+        if (!input) { return null; }
+
+        return new Intl.NumberFormat('{0}', { style: 'currency', currency: '{1}' }).format(input)
+    """ [ countryCode, currency ]
+
+
+{-| Format a cell value as localized DECIMAL value utilizing the Javascript `Intl.NumberFormat` class.
+
+    cellValue = 1200.35
+
+    decimalValueFormatter { countryCode = "de-DE", decimalPlaces = 1 }
+    > "1.200,4"
+
+    decimalValueFormatter { countryCode = "en-US", decimalPlaces = 0 }
+    > "1,200"
+
+-}
+decimalValueFormatter : { countryCode : String, decimalPlaces : Int } -> String
+decimalValueFormatter { countryCode, decimalPlaces } =
+    String.Interpolate.interpolate """
+        let input;
+
+        if (!value) { return null; }
+
+        if (typeof value === 'object' && value.hasOwnProperty('value')) {
+            // Group value
+            input = value.value
+        } else {
+            // Cell value
+            input = value
+        }
+
+        if (!input) { return null; }
+
+        return new Intl.NumberFormat('{0}', { style: 'decimal', maximumFractionDigits: '{1}' }).format(value)
+    """ [ countryCode, String.fromInt decimalPlaces ]
+
+
+{-| Format a cell value as localized PERCENT value utilizing the Javascript `Intl.NumberFormat` class.
+
+    cellValue = 0.15
+
+    percentValueFormatter { countryCode = "de-DE" }
+    > "15%"
+
+-}
+percentValueFormatter : { countryCode : String, decimalPlaces : Int } -> String
+percentValueFormatter { countryCode, decimalPlaces } =
+    String.Interpolate.interpolate """
+        let input;
+
+        if (!value) { return null; }
+
+        if (typeof value === 'object' && value.hasOwnProperty('value')) {
+            // Group value
+            input = value.value
+        } else {
+            // Cell value
+            input = value
+        }
+
+        if (!input) { return null; }
+
+        return new Intl.NumberFormat('{0}', { style: 'percent', maximumFractionDigits: {1} }).format(value)
+    """ [ countryCode, String.fromInt (decimalPlaces - 2) ]
```

Comment:
When editing the float value we allow certain decimal places to be edited. But when formatting as `Percent` we wanna display 2 decimal places less because it's a percentage value already.